services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_events_utils.py

```diff
@@ -245,13 +246,23 @@ async def service_remove_sidecar_proxy_docker_networks_and_volumes(
     # pylint: disable=protected-access
     scheduler_data.dynamic_sidecar.service_removal_state.mark_removed()
     await (
-        app.state.dynamic_sidecar_scheduler._scheduler.remove_service_from_observation(
+        app.state.dynamic_sidecar_scheduler._scheduler.remove_service_from_observation(  # noqa: SLF001
             scheduler_data.node_uuid
         )
     )
     task_progress.update(message="finished removing resources", percent=1)
 
 
+async def _remove_generated_secrets(
+    app: FastAPI, *, node_id: NodeID, product_name: ProductName, user_id: UserID
+) -> None:
+    # removes secrets that may have been generated when the sidecar was started
+
+    await get_api_keys_manager(app).safe_remove(
```

Comment:
IMO this violates the original design of the osparc variables. The idea was to have a single point to register the variable and an associated handler (in the spirit of e.g. web app handlers or events).  The first part, i.e. the creation, you did it that way but not the "tear-down" part. Instead you have hard coded it in here .


This will work (for now) but it is not maintainable as expected in the original design.

Reply:
In my mind, the liefaspan was associated to the lifespan of the user-service (the one that the user starts) and not of the service per se (director-v2).

I understand what you are proposing but I don't see any use case for it.
At this point I will rename it so that it does not cause further confusion